src/high.rs

```diff
@@ -0,0 +1,146 @@
+//! Utilities for emulating HKTs (over lifetimes) in Rust.
```

Comment:
This whole module could use a whole lot of more comments ;) It's likely that only you will understand it... (well, I could, and the basics makes sense from a type-theoretic POV, but it would need some sitting down).

Reply:
I thought I left comments on specific items in this module, do you think any need more in particular?